californium-core/src/main/java/org/eclipse/californium/core/coap/Option.java

```diff
@@ -199,7 +199,7 @@ public long getLongValue() {
 	 *
 	 * @param value the new value
 	 */
-	public void setValue(byte[] value) {
+	public final void setValue(byte[] value) {
```

Comment:
This is same as point 1 so lets park it for now until we have more structured solution.